src/main/java/com/google/api/codegen/discogapic/DiscoGapicInterfaceContext.java

```diff
@@ -16,49 +16,58 @@
 
 import com.google.api.codegen.config.GapicMethodConfig;
 import com.google.api.codegen.config.GapicProductConfig;
+import com.google.api.codegen.discogapic.transformer.DiscoGapicNamer;
 import com.google.api.codegen.discovery.Document;
 import com.google.api.codegen.transformer.FeatureConfig;
 import com.google.api.codegen.transformer.InterfaceContext;
-import com.google.api.codegen.transformer.ModelTypeTable;
+import com.google.api.codegen.transformer.SchemaTypeTable;
 import com.google.api.codegen.transformer.SurfaceNamer;
 import com.google.api.codegen.util.TypeTable;
 import com.google.api.tools.framework.model.Method;
 import com.google.auto.value.AutoValue;
 
+/**
+ * The context for transforming a Discovery Doc API into a view model to use for client library
+ * generation.
+ */
 @AutoValue
 public abstract class DiscoGapicInterfaceContext implements InterfaceContext {
   public static DiscoGapicInterfaceContext create(
       Document document,
       GapicProductConfig productConfig,
-      ModelTypeTable typeTable,
-      SurfaceNamer namer,
+      SchemaTypeTable typeTable,
+      DiscoGapicNamer discoGapicNamer,
+      SurfaceNamer surfaceNamer,
```

Comment:
The DiscoGapicNamer will simply render entities like the getter and setter method, whereas the existing SurfaceNamer class does more protobuf Model rendering. The SurfaceNamer is part of the interface that this class implements, so the SurfaceNamer isn't really used right now. Once gax is integrated, the SurfaceNamer will probably have methods to make all the Callable and Settings entities.